### PR TITLE
Allow dynamic modification of window size and deletion trigger

### PR DESCRIPTION
**Pull request by anand1976**
Summary:
Make the CompactOnDeletionCollectorFactory class public, and provide
methods to update the window size and deletion trigger params. These
will take effect on subsequent created SST files.

Test Plan:
Add new unit test
make check

### CODE REVIEW

File: include/rocksdb/utilities/table_properties_collectors.h [hunk 1]
```diff
@@ -11,6 +11,52 @@
...
+  //
+  // @param sliding_window_size "N"
+  // @param deletion_trigger "D"
+  CompactOnDeletionCollectorFactory(
+      size_t sliding_window_size,
+      size_t deletion_trigger) :
+          sliding_window_size_(sliding_window_size),
+          deletion_trigger_(deletion_trigger) {}
+
+  size_t sliding_window_size_;
```
Comment (ajkr) [inline include/rocksdb/utilities/table_properties_collectors.h]: should these be `std::atomic`s?

Comment (anand1976) [reply include/rocksdb/utilities/table_properties_collectors.h]: Good point. I'll change them to std::atomic

File: include/rocksdb/utilities/table_properties_collectors.h [hunk 2]
```diff
@@ -5,12 +5,59 @@
...
+// entries in any "N" consecutive entires.
+class CompactOnDeletionCollectorFactory
+    : public TablePropertiesCollectorFactory {
+ public:
+  virtual ~CompactOnDeletionCollectorFactory() {}
+
+  virtual TablePropertiesCollector* CreateTablePropertiesCollector(
+      TablePropertiesCollectorFactory::Context context) override;
+
+  // Change the value of sliding_window_size "N"
```
Comment (ajkr) [inline include/rocksdb/utilities/table_properties_collectors.h]: is there a way to use this as a kill-switch? e.g., by setting sliding_window_size to zero or negative? I recall that was something Mikhail mentioned to me.

Comment (anand1976) [reply include/rocksdb/utilities/table_properties_collectors.h]: Oh, it didn't come up in our conversation, but that sounds like a good idea. There is currently no way to turn it off after opening the DB. Setting sliding_window_size to 0 can be the kill switch.

**Pull request closed without merging**